src/main/scala/com/normation/cfclerk/domain/AixPasswordHashAlgo.scala

```diff
@@ -0,0 +1,331 @@
+        b64from24bit(bytes( 6), bytes( 7), bytes( 8), 4, buffer)
+        b64from24bit(bytes( 9), bytes(10), bytes(11), 4, buffer)
+        b64from24bit(bytes(12), bytes(13), bytes(14), 4, buffer)
+        b64from24bit(bytes(15), bytes(16), bytes(17), 4, buffer)
+        b64from24bit(bytes(18), bytes(19), 0        , 3, buffer)
+
+        buffer
+      }
+    }
+
+    final case object SHA256 extends ShaSpec {
+      val name   = "SHA256"
+      val byteNumber = 32
+      def scb64Encode(bytes: Array[Byte], buffer: JStringBuilder): JStringBuilder = {
+        b64from24bit(bytes( 0), bytes( 1), bytes( 2), 4, buffer)
+        b64from24bit(bytes( 3), bytes( 4), bytes( 5), 4, buffer)
+        b64from24bit(bytes( 6), bytes( 7), bytes( 8), 4, buffer)
+        b64from24bit(bytes( 9), bytes(10), bytes(11), 4, buffer)
+        b64from24bit(bytes(12), bytes(13), bytes(14), 4, buffer)
+        b64from24bit(bytes(15), bytes(16), bytes(17), 4, buffer)
+        b64from24bit(bytes(18), bytes(19), bytes(20), 4, buffer)
+        b64from24bit(bytes(21), bytes(22), bytes(23), 4, buffer)
+        b64from24bit(bytes(24), bytes(25), bytes(26), 4, buffer)
+        b64from24bit(bytes(27), bytes(28), bytes(29), 4, buffer)
+        b64from24bit(bytes(30), bytes(31), 0        , 3, buffer)
+
+        buffer
+      }
+    }
+
+
+    final case object SHA512 extends ShaSpec {
+      val name   = "SHA512"
+      val byteNumber = 64
+      def scb64Encode(bytes: Array[Byte], buffer: JStringBuilder): JStringBuilder = {
+        b64from24bit(bytes( 0), bytes( 1), bytes( 2), 4, buffer)
+        b64from24bit(bytes( 3), bytes( 4), bytes( 5), 4, buffer)
+        b64from24bit(bytes( 6), bytes( 7), bytes( 8), 4, buffer)
+        b64from24bit(bytes( 9), bytes(10), bytes(11), 4, buffer)
+        b64from24bit(bytes(12), bytes(13), bytes(14), 4, buffer)
+        b64from24bit(bytes(15), bytes(16), bytes(17), 4, buffer)
+        b64from24bit(bytes(18), bytes(19), bytes(20), 4, buffer)
+        b64from24bit(bytes(21), bytes(22), bytes(23), 4, buffer)
+        b64from24bit(bytes(24), bytes(25), bytes(26), 4, buffer)
+        b64from24bit(bytes(27), bytes(28), bytes(29), 4, buffer)
+        b64from24bit(bytes(30), bytes(31), bytes(32), 4, buffer)
+        b64from24bit(bytes(33), bytes(34), bytes(35), 4, buffer)
+        b64from24bit(bytes(36), bytes(37), bytes(38), 4, buffer)
+        b64from24bit(bytes(39), bytes(40), bytes(41), 4, buffer)
+        b64from24bit(bytes(42), bytes(43), bytes(44), 4, buffer)
+        b64from24bit(bytes(45), bytes(46), bytes(47), 4, buffer)
+        b64from24bit(bytes(48), bytes(49), bytes(50), 4, buffer)
+        b64from24bit(bytes(51), bytes(52), bytes(53), 4, buffer)
+        b64from24bit(bytes(54), bytes(55), bytes(56), 4, buffer)
+        b64from24bit(bytes(57), bytes(58), bytes(59), 4, buffer)
+        b64from24bit(bytes(60), bytes(61), bytes(62), 4, buffer)
+        b64from24bit(bytes(63), 0        , 0        , 2, buffer)
```

Comment:
Kind of. It's a char accumulator from byte array, with a non standard b64 encoding (different table). En bytes are not taken in order (well, they are, but they are not in Sha-crypt, and my code started with a Sha-Crypt base, so I kept the table)